src/redisraft.c

```diff
@@ -555,6 +555,66 @@ static void handleAsking(RedisRaftCtx *rr, RedisModuleCtx *ctx)
     RedisModule_ReplyWithSimpleString(ctx, "OK");
 }
 
+static void handleClientUnblock(RedisRaftCtx *rr, RedisModuleCtx *ctx, RaftRedisCommand *cmd)
+{
+    raft_session_t id;
+    bool error = false;
+
+    if (cmd->argc != 3 && cmd->argc != 4) {
+        RedisModule_ReplyWithError(ctx, "ERR syntax error");
+        return;
+    }
+
+    int ret = RedisModule_StringToULongLong(cmd->argv[2], &id);
+    if (ret != REDISMODULE_OK) {
+        RedisModule_ReplyWithError(ctx, "ERR value is not an integer or out of range");
+        return;
+    }
+
+    if (cmd->argc == 4) {
+        size_t str_len;
+        const char *str = RedisModule_StringPtrLen(cmd->argv[3], &str_len);
+
+        if (!strcasecmp(str, "timeout")) {
+            error = false;
+        } else if (!strcasecmp(str, "error")) {
+            error = true;
+        } else {
+            RedisModule_ReplyWithError(ctx, "ERR CLIENT UNBLOCK reason should be TIMEOUT or ERROR");
+            return;
+        }
+    }
+
+    ClientState *cs = ClientStateGetById(&redis_raft, id);
+    if (!cs || cs->blocked_req == NULL) {
+        RedisModule_ReplyWithLongLong(ctx, 0);
+        return;
+    }
+
+    raft_entry_t *entry = RaftRedisSerializeTimeout(cs->blocked_req->raft_idx, error);
+    RaftReq *req = RaftReqInit(ctx, RR_CLIENT_UNBLOCK);
+    entryAttachRaftReq(rr, entry, req);
+
+    int e = raft_recv_entry(rr->raft, entry, NULL);
+    if (e != 0) {
+        RedisModule_ReplyWithLongLong(req->ctx, 0);
+        RaftReqFree(req);
```

Comment:
Did you test this path? I think this will crash if you don't call `entryDetachRaftReq()`.because `raft_entry_release()` will call `entryFreeAttachedRaftReq()` and it will try to use `req` if I'm not mistaken.

Reply:
yes, there should be a return after the RaftReqFree()

Reply:
I don't think this is enough, please take a look at other usages. Probably, it is better to "return" in error case, something like: 

```c
if (e != 0) {
    RedisModule_ReplyWithLongLong(req->ctx, 0);
    entryDetachRaftReq(rr, entry);
    raft_entry_release(entry);
    RaftReqFree(req);
    return;
}

```